src/Resolvers/Microsoft.NET.Sdk.WorkloadManifestReader/WorkloadPackId.cs

```diff
@@ -51,7 +51,7 @@ public WorkloadPackId(string id)
     internal class PackIdJsonConverter : JsonConverter<WorkloadPackId>
     {
         public override WorkloadPackId Read(ref Utf8JsonReader reader, Type typeToConvert, JsonSerializerOptions options) =>
-            new WorkloadPackId(reader.GetString() ?? string.Empty);
+            new(reader.GetString() ?? string.Empty);
```

Comment:
The rule here allows for this because it's a => method. So it's trivial to see the new() is associated with the return type of the member.

Reply:
Good thing we have semantically aware tooling :-)

Reply:
@KalleOlaviNiemitalo 
> Does .NET SDK follow the [.NET Runtime coding style](https://github.com/dotnet/runtime/blob/v8.0.0-rc.1.23419.4/docs/coding-guidelines/coding-style.md?rgh-link-date=2023-09-23T03%3A44%3A10Z)

Through my observation so far, it does not seem to enforce any well-defined coding style. The repo is an amalgam of multiple repos, each had their own style. I'm not sure of the history beyond that, but many inconsistencies also exist because multiple teams contribute and own different features within the repo too. I actually [removed the `var` suggestions](https://github.com/dotnet/sdk/commit/b5face105f7f057706917f2accf60bf054184395#diff-0947e2727d6bad8cd0ac4122f5314bb5b04e337393075bc4b5ef143b17fcbd5b) from the .editorconfig in a different PR because the usage of `var` was so inconsistent.

Reply:
FWIW, the rules here for this analyzer are very well defined, and were considered by the analyzer team to be appropriate and in-line with other 'apparency' rules we have.  The `=>` was explicitly considered, and viewed as a case where it is immediately apparent, and thus should fit in here.

If that blocks this usage from the SDK, so be it.  But i think that's a rather onerous position to take.

Reply:
@CyrusNajmabadi Maybe some of the confusion lies in the fact the same rule doesn't apply to a `return` statement. And the `=>` operator here is just removing the need for `{ return; }` from the notation. So, some people might think of it as a consolidated `return`. Here specifically, the confusion might be because the actual expression exists on a separate line, which is not necessary.

Reply:
> Maybe some of the confusion lies in the fact the same rule doesn't apply to a return statement.

Correct, because you may have an arbitrarily large and complex statement-form, and thus things are not apparent.  Having to drill into blocks/statement forms to figure things out takes away from that.  As opposed to when it's simply 'at the top level'.

Reply:
> Here specifically, the confusion might be because the actual expression exists on a separate line, which is not necessary.

Correct, and no-apparency rules dictate anything about being on the same line.  For example, if you do:

```c#
var v =
    new List<int>()
    {
        a, b, c, d
    }
```

Then that's considered fine to use 'var', even though `new List<int>` is on a different line.  

Similarly, you can do:

```
List<int> list =
    new() { ... }
```

And this is likewise apparent.  

We view `=>` as morally equivalent to `=` regardless of which lines things are on.  This follows our previous behavior in this same space (which is like 15 years old now) :)  

Blocks/statements/returns are an entire additional level of complexity, which is why we didn't feel like they should be included (and they don't match anything else we've done in similar spaces).